coralproject/talk
# Updates for off-topic, viewing options

## What does this PR do?
**1.  Updates both plugins to support a plugin_config like:** 
```
plugin_config: {
          'coral-plugin-offtopic': {
            enabled: true,
            action_style: 'link',
            use_tooltip: true
          },
          'coral-plugin-streamtopper': {
            enabled: true,
            plugins: ['coral-plugin-offtopic']
          },
          debug: false
        }
```
This allows the plugins to be configured dynamically, per page.  So, for example, off topic is included in the build, but may only be enabled on certain pages.

Plugins will look up their personal config from props.config, get appropriate values, and do appropriate things.

**2.  Updated plugin structure**
 - include constants PLUGIN_NAME and DEFAULT_CONFIG.  DEFAULT_CONFIG contains any default config necessary.  
   This is merged with config as supplied from props.

- support exported function "isEnabled" in index.js.  
   This is used when evaluating modules in getSlotElements.  This allows a plugin to be disabled, such that it is not returned for a slot.  The isEnabled function generally uses the property "enabled" as supplied, but could include other evaluations as appropriate

- added translations for all text

**3.  Off topic updates**
 - support both checkbox and link format (config.action_style)
 - support tooltip on hover/mouseenter for off-topic tag

**4.  Viewing options updates**
 - support props.title
 - pass on click to children
 - doesn't currently put itself directly in a slot.  Additional refactoring could include providing slot in config.

## How do I test this PR?

1.  update viewing options to be in a slot (say streambox)
2.  update off topic config to be enabled, or,  add plugin_config as noted above

Off topic should include checkbox under submit box, filter in viewing options, and flag on comment, just like it does now.
